sample/src/main/java/com/yalantis/ucrop/sample/SampleActivity.java

```diff
@@ -103,23 +112,28 @@ public void onClick(View v) {
                 Random random = new Random();
                 int minSizePixels = 800;
                 int maxSizePixels = 2400;
-                startCropActivity(Uri.parse(String.format(Locale.getDefault(), "https://unsplash.it/%d/%d/?random",
+                Uri uri = Uri.parse(String.format(Locale.getDefault(), "https://unsplash.it/%d/%d/?random",
                         minSizePixels + random.nextInt(maxSizePixels - minSizePixels),
-                        minSizePixels + random.nextInt(maxSizePixels - minSizePixels))));
+                        minSizePixels + random.nextInt(maxSizePixels - minSizePixels)));
+
+                if (requestMode == REQUEST_SELECT_PICTURE)
+                    startCropActivity(uri);
+                else
+                    startActivityWithFragment(uri);
```

Comment:
These 2 modes will be barely used in the app at the same time. So probably it is better to create 2 build flavors in this sample and base this choice on them,  not on request mode. Because it just makes logic more difficult, users may copy it without realizing they don't need this complexity.